elki-outlier/src/main/java/elki/outlier/isolation/IsolationForest.java

```diff
@@ -0,0 +1,366 @@
+            maxs[d] = Math.max(v, maxs[d]);
+          }
+        }
+        minmax = new double[][] { mins, maxs };
+      }
+
+      ArrayList<Integer> al = new ArrayList<Integer>();
+      for(int j = 0; j < dim; j++) {
+        if(minmax[0][j] < minmax[1][j]) {
+          al.add(j);
+        }
+      }
+
+      // Check whether any eligible attributes have been found
+      if(al.size() == 0) {
+        return;
+      }
+      else {
+        // Randomly pick an attribute and split point
+        a = al.get(r.nextInt(al.size()));
+        q = (r.nextDouble() * (minmax[1][a] - minmax[0][a])) + minmax[0][a];
+
+        // Create sub trees
+        successors = new ArrayList<>(2);
+        for(int i = 0; i < 2; i++) {
+          ArrayModifiableDBIDs tmpData = DBIDUtil.newArray();
+          for(DBIDIter iditer = data.iter(); iditer.valid(); iditer.advance()) {
+            NumberVector vector = relation.get(iditer);
+            if((i == 0) && (vector.doubleValue(a) < q)) {
+              tmpData.add(iditer);
+            }
+            if((i == 1) && (vector.doubleValue(a) >= q)) {
+              tmpData.add(iditer);
+            }
+          }
+          successors.add(new Tree(tmpData, r, height + 1, maxHeight, relation));
+        }
+      }
+    }
+
+    /**
+     * Returns path length according to algorithm.
+     */
+    protected double pathLength(NumberVector vector) {
+      if(successors == null) {
+        return c(size);
+      }
+      if(vector.doubleValue(a) < q) {
+        return successors.get(0).pathLength(vector) + 1.0;
+      }
+      else {
+        return successors.get(1).pathLength(vector) + 1.0;
+      }
+    }
+  }
+
+  @Override
+  public TypeInformation[] getInputTypeRestriction() {
+    return TypeUtil.array(TypeUtil.NUMBER_VECTOR_FIELD);
+  }
+
+  /**
+   * Parameterization class
+   * 
+   * @hidden
+   * 
+   * @param <V> Vector type
+   */
+  public static class Par implements Parameterizer {
+    /**
+     * Parameter for the number of trees
+     */
+    public static final OptionID NUM_TREES_ID = new OptionID("iforest.numtrees", "Number of trees to use.");
+
+    /**
+     * Parameter for the sub sample size
+     */
+    public static final OptionID SUBSAMPLE_SIZE_ID = new OptionID("iforest.subsample", "Sub sample size.");
+
+    /**
+     * Parameter to specify the seed to initialize Random.
+     */
+    public static final OptionID SEED_ID = new OptionID("iforest.seed", "The seed to use for initializing Random.");
+
+    protected int numTrees = 100;
+
+    protected int subsampleSize = 256;
+
+    protected RandomFactory rnd;
+
+    @Override
+    public void configure(Parameterization config) {
+      new IntParameter(NUM_TREES_ID, 100) //
+          .addConstraint(CommonConstraints.GREATER_THAN_ONE_INT) //
+          .grab(config, x -> numTrees = x);
+      new IntParameter(SUBSAMPLE_SIZE_ID, 256) //
+          .addConstraint(CommonConstraints.GREATER_THAN_ONE_INT) //
```

Comment:
Here 2 is probably still a rather meaningless minimum size, where we would expect all scores to be the same, right?